src/wordy/ast/ConditionalNode.java

```diff
@@ -72,4 +75,51 @@ public String toString() {
     protected String describeAttributes() {
         return "(operator=" + operator + ')';
     }
+
+    @Override
+    protected void doRun(EvaluationContext context) {
+        double lhsValue = lhs.evaluate(context);
+        double rhsValue = rhs.evaluate(context);
+        boolean condition;
+        switch(operator) {
+            case EQUALS:
+                condition = lhsValue == rhsValue;
+                break;
+            case LESS_THAN:
+                condition = lhsValue < rhsValue;
+                break;
+            case GREATER_THAN:
+                condition = lhsValue > rhsValue;
+                break;
+            default:
+                throw new AssertionError("Unknown operator: " + operator);
+        }
+        if(condition) {
+            ifTrue.run(context);
+        } else {
+            ifFalse.run(context);
+        }
+    }
+
+    @Override
+    public void compile(PrintWriter out) {
+        out.print("if (");
+        lhs.compile(out);
+        switch (operator) {
+            case EQUALS:
+                out.print(" == ");
+                break;
+            case LESS_THAN:
+                out.print(" < ");
+                break;
+            case GREATER_THAN:
```

Comment:
I saw a nice new Java syntax for switch. I will definitely put that in my toolkit.